### PR TITLE
Email sign-up: OB refactor

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

We need to show the non-compliant (merchandising) OB 'widget' when the email sign-up is shown. To ensure that OB is loaded early as possible in all cases but still check whether email sign-up will show, I have refactored so that we can check if a sign-up form will be shown before showing it, therefore allowing us to insert the correct OB component.
## What is the value of this and can you measure success?

The value of this is to ensure that we are meeting our contract and so they can test the impressions of the widget.
## Request for comment

@regiskuckaertz @jbreckmckye I'm messing with Outbrain so can you guys have a glance please.

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
The Outbrain part looks great to me :+1:

**Comment by gtrufitt** [comment]
I've updated this to now handle email slightly differently - use merchandising codes but still place OB into the high relevance container: https://github.com/guardian/frontend/pull/12248/files#diff-9bc737895e7827869f14c0647c206986R40

I've also handled the ad-blocker scenario https://github.com/guardian/frontend/pull/12248/files#diff-9bc737895e7827869f14c0647c206986R161

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/outbrain.js [hunk 1]
```diff
@@ -120,6 +126,30 @@ define([
             detect.adblockInUseSync();
     }
 
+    function checkEmailSignup() {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/outbrain.js]: I feel like a small improvement would have this function return a new promise on the first call and a cached promised on the second call.

**Comment by regiskuckaertz** [comment]
Let's move the widget only in the `merchandising` case and the roll this little guy out :dancers:

**Comment by gtrufitt** [comment]
Thanks @regiskuckaertz - made those changes. Really appreciate your support on this one! :+1:

**Pull request merged**